### PR TITLE
Don't use lex_state emulation on Ruby 2.5 or later

### PR DESCRIPTION
**Pull request by aycabta**
`RDoc::RipperStateLex` has `InnerStateLex` that is emulation layer about `lex_state` of parse.y, but after Ruby 2.5 or later, `Ripper` gives `lex_state` value as `state` method inside `Ripper::Filter`'s callback. On this Pull Request, uses the new feature of `Ripper` on Ruby 2.5 or later.

### CODE REVIEW

File: lib/rdoc/parser/ripper_state_lex.rb [hunk 1]
```diff
@@ -1,6 +1,9 @@
 require 'ripper'
 
 class RDoc::RipperStateLex
+  # TODO: Remove this constants after Ruby 2.4 EOL
+  RIPPER_HAS_LEX_STATE = (RUBY_VERSION >= '2.5.0')
```
Comment (hsbt) [inline lib/rdoc/parser/ripper_state_lex.rb]: Should we use `respond_to?` instead of `RUBY_VERSION` ?

Comment (hsbt) [reply lib/rdoc/parser/ripper_state_lex.rb]: or `defined?`

**Comment by aycabta** [comment]
OK, I'll use `Ripper::Filter.method_defined?(:state)`.

File: lib/rdoc/parser/ripper_state_lex.rb [hunk 2]
```diff
@@ -1,6 +1,9 @@
 require 'ripper'
 
 class RDoc::RipperStateLex
+  # TODO: Remove this constants after Ruby 2.4 EOL
+  RIPPER_HAS_LEX_STATE = Ripper::Filter.method_defined?(:state)
```
Comment (hsbt) [inline lib/rdoc/parser/ripper_state_lex.rb]: 👍

**Pull request merged**